csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -2150,26 +2321,39 @@ private predicate pathOutOfCallable0(PathNodeMid mid, ReturnPosition pos, CallCo
 
 pragma[nomagic]
 private predicate pathOutOfCallable1(
-  PathNodeMid mid, DataFlowCall call, ReturnKindExt kind, CallContext cc
+  PathNodeMid mid, DataFlowCall call, ReturnKindExt kind, CallContext cc, AccessPath ap,
+  Configuration config
 ) {
   exists(ReturnPosition pos, DataFlowCallable c, CallContext innercc |
     pathOutOfCallable0(mid, pos, innercc) and
     c = pos.getCallable() and
     kind = pos.getKind() and
-    resolveReturn(innercc, c, call)
+    resolveReturn(innercc, c, call) and
+    ap = mid.getAp() and
```

Comment:
How is this delta performing? Do we need to push `ap = mid.getAp()` and `config = mid.getConfiguration()` into `pathOutOfCallable0`?  Mutual recursion with an IPA type unfortunately still isn't always handled very well, so this turns into some needless cases in the RA pipeline.

Reply:
I haven't observed that this should be a problem. But wouldn't `pathOutOfCallable0` have the same non-linear recursion as `pathOutOfCallable1` if it was pushed in anyway?

Reply:
I will move it into `pathOutOfCallable0`.